### PR TITLE
ENH: support decimal argument in read_html #12907

### LINKED ISSUE #12907: read_html: support "decimal" argument for parsing numbers, like read_csv
read_csv has support for declaring the decimal and thousands separator.

read_html is missing the 'decimal' parameter. it'd be useful and more consistent to accept it too.

Example:
```
pd.read_html(html, thousands='.', decimal=',')
```

### PR DESCRIPTION
**Pull request by ccronca**
- [x] closes #12907 
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/html.py [hunk 1]
```diff
@@ -828,6 +833,10 @@ def read_html(io, match='.+', flavor=None, header=None, index_col=None,
         underlying parser library (e.g., the parser library will try to use
         the encoding provided by the document).
 
+    decimal : str, default '.'
+        Character to recognize as decimal point (e.g. use ',' for European
+        data).
+
```
Comment (jreback) [inline pandas/io/html.py]: add a versionadded tag here

File: pandas/io/html.py [hunk 2]
```diff
@@ -744,15 +746,18 @@ def _parse(flavor, io, match, header, index_col, skiprows,
...
         except EmptyDataError:  # empty table
             continue
     return ret
 
 
 def read_html(io, match='.+', flavor=None, header=None, index_col=None,
               skiprows=None, attrs=None, parse_dates=False,
-              tupleize_cols=False, thousands=',', encoding=None):
+              tupleize_cols=False, thousands=',', encoding=None,
+              decimal=b'.'):
```
Comment (jreback) [inline pandas/io/html.py]: ``decimal='.'``; you can't specifiy bytes.

File: doc/source/whatsnew/v0.18.2.txt [hunk 3]
```diff
@@ -79,6 +79,7 @@ Other enhancements
 - ``Index.astype()`` now accepts an optional boolean argument ``copy``, which allows optional copying if the requirements on dtype are satisfied (:issue:`13209`)
 - ``Categorical.astype()`` now accepts an optional boolean argument ``copy``, effective when dtype is categorical (:issue:`13209`)
 
+- The ``pd.read_html()`` has gained support for the ``decimal`` option (:issue:`12907`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.2.txt]: ``pd.read_html()`` (remove The)

**Comment by camilocot** [comment]
@jreback please review latest commit

**Comment by jreback** [comment]
this looks good.

Do you have a site that this would work on? (I always like to manually test things as well)

**Comment by camilocot** [comment]
@jreback I have generated some data here: http://www.googledrive.com/host/0BzJmIC7_SNINRGxSUlhaT2hBeEk

It should work with:
`df1 = pandas.read_html(url, decimal=',', thousands='.', header=1)`

**Comment by jreback** [comment]
thanks @camilocot great!

**Pull request closed without merging**